### PR TITLE
User authorization (identicon, social), comments are linked to users

### PR DESCRIPTION
**Pull request by mXaln**
It uses custom permissions, that we can remove when we chose better model.

### CODE REVIEW

File: tRecorderApi/api/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.contrib.auth.hashers import make_password
+
+
+def user_created(strategy, details, user=None, *args, **kwargs):
+    print(user.username)
+
+    if user:
+        # password = make_password("some_password")
+
+        user.is_staff = True
```
Comment (jsarabia1247) [inline tRecorderApi/api/pipeline.py]: Does the guest account need to be staff?

File: tRecorderApi/api/views/user.py [hunk 2]
```diff
@@ -1,8 +1,166 @@
...
+    def destroy(self, request, pk=None):
+        instance = self.get_object()
+        try:
+            os.remove(instance.name_audio)
+        except OSError:
+            pass
+        self.perform_destroy(instance)
+        return Response(status=status.HTTP_200_OK)
+
+    def create(self, request):
```
Comment (jsarabia1247) [inline tRecorderApi/api/views/user.py]: This method is doing a lot that looks like it should be broken down into smaller methods, maybe something like the following:

if created:
validate_profile_upload() // verifies we have all the blobs/content needed
create_profile() // converts audio, writes content to disk at appropriate location
add_user() //creates a user object once all necessary files are stored and in their proper place

I didn't spend much time thinking about the names, just trying to communicate the couple distinct operations I see. Makes all of these distinct operations testable (and they should get test)

File: tRecorderApi/api/pipeline.py (hunk 1 above)
Comment (mXaln) [reply tRecorderApi/api/pipeline.py]: is_staff is here only to differentiate identicon users from social users. We can use different property name

File: tRecorderApi/api/views/user.py (hunk 2 above)
Comment (mXaln) [reply tRecorderApi/api/views/user.py]: Fixed in latest commit

**Pull request merged**